pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -617,7 +553,10 @@ impl DeltaLayer {
     /// Obtains all keys and value references stored in the layer
     ///
     /// The value can be obtained via the [`ValueRef::load`] function.
-    pub async fn load_val_refs(&self, ctx: &RequestContext) -> Result<Vec<(Key, Lsn, ValueRef)>> {
+    pub async fn load_val_refs(
+        &self,
+        ctx: &RequestContext,
+    ) -> Result<Vec<(Key, Lsn, ValueRef<Arc<DeltaLayerInner>>)>> {
         let inner = self
             .load(LayerAccessKind::KeyIter, ctx)
```

Comment:
Noticed that both of these should not be KeyIter, there was another variant as well.

Reply:
Not that anyone was actually watching the access stats anymore.

Reply:
a7fc54358